### ISSUE TITLE
Overwolf Overlay causes graphical corruption

### ISSUE THREAD

**Issue by jellysquid3** [jellysquid3/Sodium#2862]
Overwolf's launcher seems to include some kind of in-game overlay and it will fail to restore the OpenGL context state when it returns from the render flip hook. This causes severe problems since Minecraft caches the bound resources for pipeline texture slots and does not know when the overlay software has modified them. Most commonly, the result is that terrain renders as some random color, due to the light texture slot being bound to the incorrect texture.

**Comment by jellysquid3** [comment]
I'm not in the mood to try and reverse engineer their software and figure out how we can mitigate the bug on our side. 

Unlike other overlay software (RTSS, MSI Afterburner, ASUS GPU Tweak, etc) it doesn't seem to be such a big deal if users are asked to turn it off. Frankly, after reading their website, it is difficult to understand what it even does.

So I am just going to add it to the module blacklist in the mod and focus on more important bugs.

**Comment by Heaser** [comment]
No need to reverse engineer or anything like that :)
Overwolf human person here.
We've known about this but we're unable to reproduce it, 
It would be cool if you could please either let me know how to reproduce or connect me with someone who can reproduce it, We've been trying for over a week to reproduce this now..

edit:
reproduced, working on a fix.

edit 2:
fix is being tested now. We accidently set `glActiveTexture`  twice in some cases

**Comment by jellysquid3** [comment]
I'd be happy to remove the module from the blacklist if you can resolve these problems. We don't have a computer that can reproduce the problem but we do have API traces that show the problematic behavior.

But, it would be nice if we could look at the module's version to determine if an old version is used. Right now Overwolf's `gep_minecraft.dll` module does not contain any versioning information.

**Comment by jellysquid3** [comment]
At least, if you can tell us where your module hooks into Minecraft specifically, we could try clearing the `GL_ACTIVE_TEXTURE` state only when your module is loaded.

**Comment by Heaser** [comment]
There is no way you should do literally anything to fix our bug. 

I'll just update once a fixed version is deployed

Also, you should look at owclient.dll and not gep_minecraft.dll

edit: we are now gradually rolling out the version with the fix

**Comment by jellysquid3** [comment]
I'm reverting https://github.com/CaffeineMC/sodium/commit/e7ea6f7dd5207c8fd0dac584a1134a16025a85de for the time being since you have said it is fixed.

**Comment by Heaser** [comment]
Thank you but just an fyi we haven't finished the deployment of the version with the fix.

It's at about 60% deployed

Also if you encounter any other compatibility issue like this feel free to let me know discord or here or wherever. (IMS also contacted me about it but I wasn't able to reproduce this until I saw in a different issue that I had to go to the nether)

thanks

edit 1:
Fix now available for everyone.
